src/Synapse/Synapse/Commands/DataPlaneCommands/AccessControl/NewAzureSynapseRoleAssignment.cs

```diff
@@ -97,6 +96,20 @@ public class NewAzureSynapseRoleAssignment : SynapseRoleCmdletBase
         [ValidateNotNullOrEmpty]
         public string ObjectId { get; set; }
 
+        // Compared with Remove-AzSynapseRoleAssignment and Get-AzSynapseRoleAssignment, no need to specify roleAssignment, it is created as
+        // random uuid. Hence unnecessary to specify the ParameterSetName
+        [Parameter(ValueFromPipelineByPropertyName = false, Mandatory = false, HelpMessage = HelpMessages.WorkspaceItemType)]
+        [ValidateNotNullOrEmpty]
+        public WorkspaceItemType ItemType { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = false, Mandatory = false, HelpMessage = HelpMessages.WorkspaceItem)]
+        [ValidateNotNullOrEmpty]
+        public string Item { get; set; }
+
+        [Parameter(ValueFromPipelineByPropertyName = false, Mandatory = false, HelpMessage = HelpMessages.WorkspacePrincipleType)]
+        [ValidateNotNullOrEmpty]
+        public string PrincipleType { get; set; }
```

Comment:
>PrincipleType [](start = 22, length = 13)

In the Swagger spec(SDK), we use the word `principal` rather than `principle`. Is it more natural to use `principle`?

Reply:
1. do we also need to update objectid to principleid?
2. if we use 'principle' as principletype, then what we should do with objectid?

an option may be: keep 'objectid' unchanged, change 'prinicpletype' to 'principle'

Reply:
1. we don't need to change `objectid` because it already has an alias `principalId`
2. I am not saying change `principleType` to `principle`. I am concerned about the spelling, princip**LE** or pincip**AL**?